v1.1/parallel-statement-execution.md

```diff
@@ -0,0 +1,127 @@
+
+In our sample scenario, the transaction would be as follows:
+
+~~~ sql
+> BEGIN;
+> UPDATE users SET last_name = 'Smith' WHERE id = 1 RETURNING NOTHING;
+> UPDATE favorite_movies SET movies = 'The Matrix' WHERE user_id = 1 RETURNING NOTHING;
+> UPDATE favorite_songs SET songs = 'All this time' WHERE user_id = 1 RETURNING NOTHING;
+> COMMIT;
+~~~
+
+In this case, because the `UPDATE` statements within the transaction are independent of each other, they can be executed in parallel without affecting the results. The `COMMIT` statement is the barrier statement and is executed sequentially. A barrier statement is executed only after all the parallel statements preceding it have finished executing.
+
+The following conceptual diagram shows how the transaction is executed sequentially and in parallel. The diagram also shows how executing statements in parallel reduces the aggregate latency.
+
+<img src="{{ 'images/Parallel_Statement_Normal_Execution.png' | relative_url }}" alt="CockroachDB Parallel Statement Execution" style="border:1px solid #eee;max-width:100%" />
+
+### Perceived delay in execution of barrier statements 
+
+As stated earlier, the server executes a barrier statement only after all the preceding parallel statements have finished executing. So it may seem as if the barrier statement is taking longer to execute, but it is waiting on the parallel statements. Even then, the total time required for parallel execution of statements followed by the sequential execution of the barrier statement should be less than the time required for the sequential execution of all statements. 
+
+Referring to the previous diagram, the server executes all `UPDATE` statements to completion before executing `COMMIT`. Hence it might seem as if `COMMIT` is taking longer to execute, but it is, in fact, waiting on the `UPDATE` statements to finish executing.
+
+### Error message mismatch
+
+In case of parallel statement execution, you might see a mismatch in the error message being displayed and the statement being executed. 
+
+Because the `RETURNING NOTHING` clause is used with parallel statements, CockroachDB cannot return the error for the parallel statement. The server reports the error message for the parallel statement after the next barrier statement. 
+
+On encountering the error, CockroachDB stops the execution of all parallel statements and [aborts and retries the transaction](transactions.html#error-handling). So even if an error message mismatch occurs, the result is the same as aborting the transaction.
+
+Revisiting our sample scenario, suppose an error occurs while executing the second `UPDATE` statement. In sequential execution, the transaction is aborted and the error message is sent immediately. However, in case of parallel execution, the transaction is aborted and the error message is sent after `COMMIT`. The following diagram illustrates the concept of error message mismatch:
```

Comment:
I think this paragraph actually has the key points. We might want to start with this and condense some of the rest:

> With sequential execution, as soon as an error happens, the transaction is aborted and an error message is sent to the client. However, with parallel execution, this happens not when the error is encountered but rather after the next barrier statement. This can result in the client receiving an error message that doesn't match the statement being executed. The following diagram illustrates this concept:
>
> DIAGRAM